rep-0154.rst

```diff
@@ -0,0 +1,104 @@
+REP: 154
+Title: test_abi support for ROS distribution files
+Author: Jose Luis Rivero <jrivero@openrobotics.org>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 21-Nov-2018
+Post-History: 21-Nov-2019
+
+
+.. contents::
+
+Abstract
+========
+This REP updates the specification of the ROS distribution files facilitated in
+the building, packaging, testing and documenting process.
+
+The intention is to annotate a ROS distribution with more information,
+particularly with the option of running an ABI/API analysis for pull requests
+and devel jobs.
+
+This REP is an extension to the file format originally defined in REP 143[1]_
+and extended in REP 153[2]_. It currently does not repeat the content of REPs
+143 and 153 but only states the differences.
+
+
+Motivation
+==========
+
+Since the creation of ROS1 some developers and users have been looking into
+improving the stability of the different updates that reach the set of
+ROS packages. Two aspects of the stability are the respect of the current
+API and, extending this first one, the stability of the ABI.
+
+Traditionally the ROS policy related to releases[3]_ does not impose any
+restriction to changes in the application or binary interface of the
+ROS packages. This lack of restriction applies to changes inside the
+same ROS distribution or changes between different ROS distributions.
+
+The inclusion of an option for testing the ABI/API changes in ROS
+packages can cover at least two scenarios:
+
+1. Run an API/ABI analysis together with changes coming from
+   pull requests. Currently the option to use CI in the pull request
+   is covered by the `test_pull_requests` (as described in REP 143[1]_)
+
+1. Run an API/ABI analysis of changes that appear in the source code
+   of ROS packages. This would be part of the CI cycle present in the devel
+   scripts.
+
+The original target to compare both use cases and to be taken as the stable
+API/ABI are the latest stable released packages for each ROS distribution
+under testing.
+
+Specification
+=============
+
+Distribution file
+-----------------
+
+* repositories
+
+  * source
+
+    * test_abi: an optional boolean (default false) to run an API/ABI analsys
+      of the code in the pull request against the latest released packages in
+      devel jobs. If ``test_pull_requests`` is enabled, run the analysis as
+      part of the CI integrated in pull requests.
+
```

Comment:
The REP needs to mention the `version` field of the distribution file. See the above comment for the question if we need to bump it.